libtiny_tui/src/msg_area/mod.rs

```diff
@@ -159,6 +161,12 @@ impl MsgArea {
         let line_height = self.line_buf.rendered_height(self.width);
         self.lines
             .push(mem::replace(&mut self.line_buf, Line::new()));
+        // Remove first element because we went over the maximum lines
+        if let Some(max_lines) = self.max_lines {
+            if self.lines.len() > max_lines {
+                self.lines.remove(0);
```

Comment:
Why not use `VecDeque` here? This potentially (depending of buffer size) shifts thousands of elements.